### PR TITLE
Build disk images in CI

### PR DESCRIPTION
**Pull request by aghecenco**

### CODE REVIEW

**Comment by aghecenco** [comment]
Needs https://github.com/rust-vmm/rust-vmm-container/pull/34 and a new `rust-vmm-container` image to be clean.

File: .buildkite/deps-pipeline.yml
Comment (alexandruag) [inline .buildkite/deps-pipeline.yml:4]: Is this script busybox-specific? If so, `make_busybox_image` might be a more accurate name/description.

File: resources/kernel/make_kernel.sh
Comment (alexandruag) [inline resources/kernel/make_kernel.sh:158]: Hmm, why is this added as an extra param versus how things were before?

File: resources/kernel/make_kernel_image_deb.sh
Comment (alexandruag) [inline resources/kernel/make_kernel_image_deb.sh:106]: Hmmm, what are the deb packages for? It seems there are variables for things such as the kernel version, but then we have these links hard-coded here.

File: .buildkite/deps-pipeline.yml
Comment (aghecenco) [reply .buildkite/deps-pipeline.yml:4]: Yup, it builds kernels witth busybox initramfs'es. I'll re-rename it.

File: resources/kernel/make_kernel.sh
Comment (aghecenco) [reply resources/kernel/make_kernel.sh:158]: Because this function can be recycled to build `.deb`s with `make deb-pkg`. So the `target` is no longer a variable of the kernel format.

File: resources/kernel/make_kernel_image_deb.sh
Comment (aghecenco) [reply resources/kernel/make_kernel_image_deb.sh:106]: It's the kernel image and modules. Even though we don't use the kernel shipped with the `linux-image` deb, Ubuntu still needs a bunch of modules in the `linux-modules` deb (and `System.map` for some reason) to work properly.

Comment (aghecenco) [reply resources/kernel/make_kernel_image_deb.sh:106]: As for the kernel version: we don't anywhere to download the *exact* kernel Ubuntu 20.04 ships with, so we instead build one with the same major.minor and thankfully it works fine. Turns out, you can't find *everything* on the Internet :(

Comment (alexandruag) [reply resources/kernel/make_kernel_image_deb.sh:106]: Hmm, aren't the modules and System.map available after building the kernel from sources as well? Also, does it make sense to use a kernel config where the required things are compiled as part of the image so we don't need to load modules?

Comment (aghecenco) [reply resources/kernel/make_kernel_image_deb.sh:106]: I think they can be built with other `make` recipes, but it takes way longer than it does to download them, and I'm not sure how to all the pieces come together to make a full Ubuntu. I went with this 2-step solution because it did the trick, and doesn't require multiple (re)compilations, but if you know a quicker/simpler way to get the working OS, we should go with it.

Comment (alexandruag) [reply resources/kernel/make_kernel_image_deb.sh:106]: I don't TBH, but the current one was/is a bit hard to understand and I was wondering if we can simplify things. I guess one way forward here is to go as things are until we have the initial devices merged and exercised by some tests as part of the CI etc, and then take a better look at the system itself, and at the very least add some more documentation if no other changes are required or important for clarity and reuse. What do other folks think here?

File: docs/getting-started.md
Comment (lauralt) [inline docs/getting-started.md:91]: This path should be updated.

File: resources/disk/make_rootfs.sh
Comment (lauralt) [inline resources/disk/make_rootfs.sh:44]: Looks like we don't have a `h` option anymore and can be removed from here.

File: resources/kernel/make_kernel_image_deb.sh
Comment (lauralt) [reply resources/kernel/make_kernel_image_deb.sh:106]: We can go with how things are right now (the documentation is pretty clear) and, like you've said, think about simplifications after devices are tested by the CI. Also there is some duplicate code in the kernel scripts that could probably be avoided, but for now I'm okay with the PR.

File: tests/test_run_reference_vmm.py
Comment (lauralt) [inline tests/test_run_reference_vmm.py:42]: nit: can we update the `cmdline` with the suggestion from [here](https://github.com/rust-vmm/vmm-reference/pull/41#discussion_r534318525)?

**Review by alexandruag — COMMENTED** [review]
Looks like we can merge this after addressing the remaining comments.

**Review by lauralt — APPROVED** [review]

**Review by alexandruag — APPROVED** [review]

**Pull request merged**
